code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -98,11 +124,295 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == "zombie" || name == "skeleton")
```

Comment:
No, I assume you're using that to make sure you're not giving a number to unique zombies like on the gym vault, righ?

Do `name == initial(name)` so unique names don't get numbers but if somebody adds a subtype with different name they don't get issues with this.